runtime/service.go

```diff
@@ -48,20 +57,72 @@ import (
 	"github.com/firecracker-microvm/firecracker-containerd/eventbridge"
 	"github.com/firecracker-microvm/firecracker-containerd/internal"
 	"github.com/firecracker-microvm/firecracker-containerd/proto"
+	"github.com/firecracker-microvm/firecracker-containerd/runtime/firecrackeroci"
 )
 
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
 const (
 	defaultVsockPort     = 10789
 	supportedMountFSType = "ext4"
+
+	vmIDEnvVarKey = "FIRECRACKER_VM_ID"
+
+	varRunDir                  = "/var/run/firecracker-containerd"
+	firecrackerSockName        = "firecracker.sock"
+	firecrackerLogFifoName     = "fc-logs.fifo"
+	firecrackerMetricsFifoName = "fc-metrics.fifo"
+
+	// TODO these strings are hardcoded throughout the containerd codebase, it may
+	// be worth sending them a PR to define them as constants accessible to shim
+	// implementations like our own
+	shimAddrFileName = "address"
+	shimLogFifoName  = "log"
+	ociConfigName    = "config.json"
 )
 
 // implements shimapi
 type service struct {
 	server        *ttrpc.Server
 	id            string
+	vmID          string
 	publish       events.Publisher
 	eventExchange *exchange.Exchange
 
+	// vmIDDir holds files, sockets and FIFOs scoped to a single given VM
+	vmIDDir string
+
+	// bundleDir is created by containerd and contains configuration scoped to a
+	// single given container
+	bundleDir string
+
+	// shimAddrFilePath is the path to the shim address file as found in the vmIDDir
```

Comment:
What is the address file?  A text file containing the path to the AF_UNIX socket for the shim?